From bd3a07f6e01c52b22e369b3d3e2ed0807829c195 Mon Sep 17 00:00:00 2001
From: Maxime Ripard <maxime@cerno.tech>
Date: Thu, 22 Apr 2021 10:45:37 +0200
Subject: [PATCH 387/552] clk: requests: Dereference the request pointer after
 the check

The current code will first dereference the req pointer and then test if
it's NULL, resulting in a NULL pointer dereference if req is indeed
NULL. Reorder the test and derefence to avoid the issue

Signed-off-by: Maxime Ripard <maxime@cerno.tech>
---
 drivers/clk/clk.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/clk.c b/drivers/clk/clk.c
index 5a659cd1fccf..438c5f8f0b1e 100644
--- a/drivers/clk/clk.c
+++ b/drivers/clk/clk.c
@@ -2515,10 +2515,11 @@ EXPORT_SYMBOL_GPL(clk_request_start);
  */
 void clk_request_done(struct clk_request *req)
 {
-	struct clk_core *core = req->clk->core;
+	struct clk_core *core;
 
 	if (!req)
 		return;
+	core = req->clk->core;
 
 	clk_prepare_lock();
 
-- 
2.33.0

